p2p/server/server.go

```diff
@@ -147,26 +228,51 @@ func (s *Server) streamHandler(stream network.Stream) {
 	}
 }
 
+func (s *Server) trackClientLatency(start time.Time, err error) {
+	if s.metrics != nil {
+		if err != nil {
+			s.metrics.clientLatencyFailure.Observe(time.Since(start).Seconds())
+		} else {
+			s.metrics.clientLatency.Observe(time.Since(start).Seconds())
+		}
+	}
```

Comment:
NIT:

```suggestion
	switch {
	case s.metrics == nil:
		return
	case err != nil:
		s.metrics.clientLatencyFailure.Observe(time.Since(start).Seconds())
	default:
		s.metrics.clientLatency.Observe(time.Since(start).Seconds())
	}
```